ui/app/src/components/PublishOnDemandWidget/PublishOnDemandWidget.tsx

```diff
@@ -153,6 +157,13 @@ export function PublishOnDemandWidget(props: PublishOnDemandWidgetProps) {
     }
   };
 
+  useEffect(() => {
+    fnRefs.current.onSubmittingAndOrPendingChange?.({
+      hasPendingChanges: mode === 'studio' ? publishStudioFormValid : publishGitFormValid,
```

Comment:
Don't see how `valid` is suitable to identify if there are changes.

Reply:
Valid is true when the form has content (so it can be published), if there is no content in the form (is blank) then valid becomes false